File: csharpcore/GildedRoseTests/TexttestFixture.cs

using GildedRoseKata;

using System;
using System.Collections.Generic;

namespace GildedRoseTests
{
    public static class TexttestFixture
    {
        public static void Main(string[] args)
        {
            Console.WriteLine("TextTest");

            IList<Item> Items = new List<Item>{
                new Item {Name = "Vest", SellIn = 10, Quality = 20},
                new Item {Name = "Aged Brie", SellIn = 2, Quality = 0},
                new Item {Name = "Juice", SellIn = 5, Quality = 7},
                new Item {Name = "Diamond", SellIn = 0, Quality = 80},
                new Item {Name = "Diamond", SellIn = -1, Quality = 80},
                new Item
                {
                    Name = "Backstage passes",
                    SellIn = 15,
                    Quality = 20
                },
                new Item
                {
                    Name = "Backstage passes",
                    SellIn = 10,
                    Quality = 49
                },
                new Item
                {
                    Name = "Backstage passes",
                    SellIn = 5,
                    Quality = 49
                },
                // this conjured item does not work properly yet
                new Item {Name = "Bio Cake", SellIn = 3, Quality = 6}
            };

            var app = new GildedRose(Items);

            int days = 31;
            if (args.Length > 0)
            {
                days = int.Parse(args[0]) + 1;
            }

            for (var i = 0; i < days; i++)
            {
                Console.WriteLine("-------- day " + i + " --------");
                Console.WriteLine("name, sellIn, quality");
                for (var j = 0; j < Items.Count; j++)
                {
                    System.Console.WriteLine(Items[j].Name + ", " + Items[j].SellIn + ", " + Items[j].Quality);
                }
                Console.WriteLine("");
                app.UpdateQuality();
            }
        }
    }
}

File: js-jest/test/texttest_fixture.js

const { Shop, Item } = require("../src/gilded_rose");

const items = [
  new Item("Vest", 10, 20),
  new Item("Aged Brie", 2, 0),
  new Item("Juice", 5, 7),
  new Item("Diamond", 0, 80),
  new Item("Diamond", -1, 80),
  new Item("Backstage passes", 15, 20),
  new Item("Backstage passes", 10, 49),
  new Item("Backstage passes", 5, 49),

  // This Bio item does not work properly yet
  new Item("Bio Cake", 3, 6),
];

const days = Number(process.argv[2]) || 2;
const gildedRose = new Shop(items);

console.log("TextTest");
for (let day = 0; day < days; day++) {
  console.log(`\n-------- day ${day} --------`);
  console.log("name, sellIn, quality");
  items.forEach(item => console.log(`${item.name}, ${item.sellIn}, ${item.quality}`));
  gildedRose.updateQuality();
}
